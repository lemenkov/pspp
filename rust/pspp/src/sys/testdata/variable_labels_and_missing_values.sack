# File header.
"$FL2"; s60 "$(#) SPSS DATA FILE PSPP synthetic test file";
2; # Layout code
28; # Nominal case size
0; # Not compressed
0; # Not weighted
1; # 1 case.
100.0; # Bias.
"05 Jan 11"; "20:53:52";
"PSPP synthetic test file: "; i8 244; i8 245; i8 246; i8 248; s34 "";
i8 0 *3;

# Numeric variable, no label or missing values.
2; 0; 0; 0; 0x050800 *2; s8 "NUM1";

# Numeric variable, variable label.
2; 0; 1; 0; 0x050800 *2; s8 "NUM2";
32; "Numeric variable 2's label ("; i8 249; i8 250; i8 251; ")";

# Numeric variable, one missing value.
2; 0; 0; 1; 0x050800 *2; s8 "NUM3";
1.0;

# Numeric variable, variable label and missing value.
2; 0; 1; 1; 0x050800 *2; s8 "NUM4";
30; "Another numeric variable label"; i8 0 * 2;
1.0;

# Numeric variable, two missing values.
2; 0; 0; 2; 0x050800 *2; s8 "NUM5"; 1.0; 2.0;

# Numeric variable, three missing values.
2; 0; 0; 3; 0x050800 *2; s8 "NUM6"; 1.0; 2.0; 3.0;

# Numeric variable, range of missing values.
2; 0; 0; -2; 0x050800 *2; s8 "NUM7"; 1.0; 3.0;

# Numeric variables, range of missing values plus discrete value.
2; 0; 0; -3; 0x050800 *2; s8 "NUM8"; 1.0; 3.0; 5.0;
2; 0; 0; -3; 0x050800 *2; s8 "NUM9"; 1.0; HIGHEST; -5.0;
2; 0; 0; -3; 0x050800 *2; "NUM"; i8 192; i8 200; i8 204; i8 209; i8 210;
LOWEST; 1.0; 5.0;

# String variable, no label or missing values.
2; 4; 0; 0; 0x010400 *2; s8 "STR1";

# String variable, variable label.
2; 4; 1; 0; 0x010400 *2; s8 "STR2";
25; "String variable 2's label"; i8 0 * 3;

# String variable, one missing value.
2; 4; 0; 1; 0x010400 *2; s8 "STR3"; s8 "MISS";

# String variable, variable label and missing value.
2; 4; 1; 1; 0x010400 *2; s8 "STR4";
29; "Another string variable label"; i8 0 * 3;
s8 "OTHR";

# String variable, two missing values.
2; 4; 0; 2; 0x010400 *2; s8 "STR5"; s8 "MISS"; s8 "OTHR";

# String variable, three missing values.
2; 4; 0; 3; 0x010400 *2; s8 "STR6"; s8 "MISS"; s8 "OTHR"; s8 "MORE";

# Long string variable, one missing value.
# (This is not how SPSS represents missing values for long strings--it
# uses a separate record as shown later below--but old versions of PSPP
# did use this representation so we continue supporting it for backward
# compatibility.
2; 11; 0; 1; 0x010b00 *2; s8 "STR7"; "first8by";
2; -1; 0; 0; 0; 0; s8 "";

# Long string variables that will have missing values added with a
# later record.
2; 9; 0; 0; 0x010900 *2; s8 "STR8";
2; -1; 0; 0; 0; 0; s8 "";
2; 10; 0; 0; 0x010a00 *2; s8 "STR9";
2; -1; 0; 0; 0; 0; s8 "";
2; 11; 0; 0; 0x010b00 *2; s8 "STR10";
2; -1; 0; 0; 0; 0; s8 "";

# Long string variable, value label.
2; 25; 1; 0; 0x011900 *2; s8 "STR11"; 14; "25-byte string"; i8 0 * 2;
( 2; -1; 0; 0; 0; 0; s8 ""; ) * 2;
# Variable label fields on continuation records have been spotted in system
# files created by "SPSS Power Macintosh Release 6.1".
2; -1; 1; 0; 0; 0; s8 ""; 20; "dummy variable label";

# Machine integer info record.
7; 3; 4; 8; 1; 2; 3; -1; 1; 1; ENDIAN; 1252;

# Machine floating-point info record.
7; 4; 8; 3; SYSMIS; HIGHEST; LOWEST;

# Long string variable missing values record.
7; 22; 1; COUNT (
# One missing value for STR8.
COUNT("STR8"); i8 1; 8; "abcdefgh";

# Two missing values for STR9.
COUNT("STR9"); i8 2; 8; "abcdefgh"; "01234567";

# Three missing values for STR9.
COUNT("STR10"); i8 3; 8; "abcdefgh"; "01234567"; "0       ";
);

# Character encoding record.
7; 20; 1; 12; "windows-1252";

# Dictionary termination record.
999; 0;

# Data.
1.0; 2.0; 3.0; 4.0; 5.0; 6.0; 7.0; 8.0; 9.0; 10.0;
s8 "abcd"; s8 "efgh"; s8 "ijkl"; s8 "mnop"; s8 "qrst"; s8 "uvwx";
s16 "yzABCDEFGHI"; s16 "JKLMNOPQR"; s16 "STUVWXYZ01";
s16 "23456789abc"; s32 "defghijklmnopqstuvwxyzABC";
