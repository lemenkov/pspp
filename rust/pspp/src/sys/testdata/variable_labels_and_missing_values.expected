╭──────────────────────┬────────────────────────╮
│       Created        │    05-JAN-2011 20:53:52│
├──────────────────────┼────────────────────────┤
│Writer Product        │PSPP synthetic test file│
│       Version        │1.2.3                   │
├──────────────────────┼────────────────────────┤
│       Compression    │None                    │
│       Number of Cases│                       1│
╰──────────────────────┴────────────────────────╯

╭─────────┬──────────────────────────────╮
│Label    │PSPP synthetic test file: ôõöø│
│Variables│                            21│
╰─────────┴──────────────────────────────╯

╭────────────────────────────────┬────────┬────────────────────────────────┬─────────────────┬─────┬─────┬─────────┬────────────┬────────────┬───────────────────────────╮
│                                │Position│              Label             │Measurement Level│ Role│Width│Alignment│Print Format│Write Format│       Missing Values      │
├────────────────────────────────┼────────┼────────────────────────────────┼─────────────────┼─────┼─────┼─────────┼────────────┼────────────┼───────────────────────────┤
│num1                            │       1│                                │                 │Input│    8│Right    │F8.0        │F8.0        │                           │
│Numeric variable 2's label (ùúû)│       2│Numeric variable 2's label (ùúû)│                 │Input│    8│Right    │F8.0        │F8.0        │                           │
│num3                            │       3│                                │                 │Input│    8│Right    │F8.0        │F8.0        │1                          │
│Another numeric variable label  │       4│Another numeric variable label  │                 │Input│    8│Right    │F8.0        │F8.0        │1                          │
│num5                            │       5│                                │                 │Input│    8│Right    │F8.0        │F8.0        │1; 2                       │
│num6                            │       6│                                │                 │Input│    8│Right    │F8.0        │F8.0        │1; 2; 3                    │
│num7                            │       7│                                │                 │Input│    8│Right    │F8.0        │F8.0        │1 THRU 3                   │
│num8                            │       8│                                │                 │Input│    8│Right    │F8.0        │F8.0        │1 THRU 3; 5                │
│num9                            │       9│                                │                 │Input│    8│Right    │F8.0        │F8.0        │1 THRU HIGH; -5            │
│numàèìñò                        │      10│                                │                 │Input│    8│Right    │F8.0        │F8.0        │LOW THRU 1; 5              │
│str1                            │      11│                                │Nominal          │Input│    4│Left     │A4          │A4          │                           │
│String variable 2's label       │      12│String variable 2's label       │Nominal          │Input│    4│Left     │A4          │A4          │                           │
│str3                            │      13│                                │Nominal          │Input│    4│Left     │A4          │A4          │"MISS"                     │
│Another string variable label   │      14│Another string variable label   │Nominal          │Input│    4│Left     │A4          │A4          │"OTHR"                     │
│str5                            │      15│                                │Nominal          │Input│    4│Left     │A4          │A4          │"MISS"; "OTHR"             │
│str6                            │      16│                                │Nominal          │Input│    4│Left     │A4          │A4          │"MISS"; "OTHR"; "MORE"     │
│str7                            │      17│                                │Nominal          │Input│   11│Left     │A11         │A11         │"first8by"                 │
│str8                            │      18│                                │Nominal          │Input│    9│Left     │A9          │A9          │"abcdefgh"                 │
│str9                            │      19│                                │Nominal          │Input│   10│Left     │A10         │A10         │"abcdefgh"; "01234567"     │
│str10                           │      20│                                │Nominal          │Input│   11│Left     │A11         │A11         │"abcdefgh"; "01234567"; "0"│
│25-byte string                  │      21│25-byte string                  │Nominal          │Input│   25│Left     │A25         │A25         │                           │
╰────────────────────────────────┴────────┴────────────────────────────────┴─────────────────┴─────┴─────┴─────────┴────────────┴────────────┴───────────────────────────╯

╭────┬────┬────────────────────────────────┬────┬──────────────────────────────┬────┬────┬────┬────┬────┬────────┬────┬─────────────────────────┬────┬─────────────────────────────┬────┬────┬───────────┬─────────┬──────────┬───────────┬─────────────────────────╮
│Case│num1│Numeric variable 2's label (ùúû)│num3│Another numeric variable label│num5│num6│num7│num8│num9│numàèìñò│str1│String variable 2's label│str3│Another string variable label│str5│str6│    str7   │   str8  │   str9   │   str10   │      25-byte string     │
├────┼────┼────────────────────────────────┼────┼──────────────────────────────┼────┼────┼────┼────┼────┼────────┼────┼─────────────────────────┼────┼─────────────────────────────┼────┼────┼───────────┼─────────┼──────────┼───────────┼─────────────────────────┤
│1   │1.00│                            2.00│3.00│                          4.00│5.00│6.00│7.00│8.00│9.00│   10.00│abcd│efgh                     │ijkl│mnop                         │qrst│uvwx│yzABCDEFGHI│JKLMNOPQR│STUVWXYZ01│23456789abc│defghijklmnopqstuvwxyzABC│
╰────┴────┴────────────────────────────────┴────┴──────────────────────────────┴────┴────┴────┴────┴────┴────────┴────┴─────────────────────────┴────┴─────────────────────────────┴────┴────┴───────────┴─────────┴──────────┴───────────┴─────────────────────────╯
